instance/tasks.py

```diff
@@ -93,15 +95,22 @@ def make_appserver_active(appserver_id, active=True, deactivate_others=False):
     """
     Mark an AppServer as active or inactive.
     """
+    success = False
     appserver = OpenEdXAppServer.objects.get(pk=appserver_id)
     logger.info('%s %s: ID=%s', "Activating" if active else "Deactivating", appserver, appserver_id)
-    appserver.make_active(active=active)
-    if active and deactivate_others:
-        other_appservers = appserver.instance.appserver_set.filter(_is_active=True).exclude(pk=appserver_id)
-        for appserver_to_deactivate in other_appservers:
-            logger.info('Deactivating %s [%s]', appserver_to_deactivate, appserver_to_deactivate.id)
-            appserver_to_deactivate.make_active(active=False)
 
+    if appserver.server.status == Server.Status.Ready:
+        appserver.make_active(active=active)
+        success = True
+        if active and deactivate_others:
+            other_appservers = appserver.instance.appserver_set.filter(_is_active=True).exclude(pk=appserver_id)
+            for appserver_to_deactivate in other_appservers:
+                logger.info('Deactivating %s [%s]', appserver_to_deactivate, appserver_to_deactivate.id)
+                appserver_to_deactivate.make_active(active=False)
+    else:
+        success = False
```

Comment:
@UmanShahzad, what if `active` was passed in as `False`, i.e. when deactivating the app server? It does not seem appropriate to not deactivate when the appserver.server is not ready.

Reply:
With the code as you currently have it, this will still happen -- deactivating the appserver only happens under the condition of the server status being ready.

The only case I can think of right now where the VM is not 'Ready' but is still active is in the 'Unknown' state, and in that state we don't want to deactivate anyway right, because it's dangerous?

Reply:
@UmanShahzad , I have changed the logic around a bit. can you take a look at it again?

Reply:
@afzaledx I took a look, and it will do as you said, but is it what we want? See my last paragraph above.

> The only case I can think of right now where the VM is not 'Ready' but is still active is in the 'Unknown' state, and in that state we don't want to deactivate anyway right, because it's dangerous?